src/main/java/uk/gov/hmcts/reform/pip/data/management/service/ValidationService.java

```diff
@@ -146,24 +150,29 @@ public void validateBody(MultipartFile file) {
      *
      * @param jsonPayload The JSON body to validate.
      */
-    public void validateBody(String jsonPayload, ListType listType) {
+    public void validateBody(String jsonPayload, HeaderGroup headers) {
+        Map<String, String> propertiesMap = headers.getAppInsightsHeaderMap();
```

Comment:
Nothing wrong with the way you construct the headerGroup map, but there is cleaner way to do that. 

If you want to convert the key of your map to upper case, can try the guava library which is good with this type of case conversion:

CaseFormat.LOWER_CAMEL.to(CaseFormat.UPPER_UNDERSCORE, {{ReplaceWithStringToConvert}})

I think the following might be all you need. It is you which way you prefer:

        ObjectMapper mapper = new ObjectMapper();
        mapper.findAndRegisterModules();
        Map<String, Object> map = mapper.convertValue(headerGroup, new TypeReference<>() {});
        Map<String, String> newMap = map.entrySet()
            .stream()
            .filter(e -> e.getValue() != null)
            .collect(Collectors.toMap(
                e -> CaseFormat.LOWER_CAMEL.to(CaseFormat.UPPER_UNDERSCORE, e.getKey()),
                e -> e.getValue().toString()
            ));

Reply:
Nice that looks good! Will update the branch and test it :)

Reply:
Discussed on slack and leaving it how it is